Assets/HoloToolkit/UX/Scripts/Buttons/ButtonSounds.cs

```diff
@@ -0,0 +1,106 @@
+﻿//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using UnityEngine;
+
+namespace HoloToolkit.Unity.Buttons
+{
+    /// <summary>
+    /// A convenient way to play sounds in response to button actions / states
+    /// </summary>
+    [RequireComponent(typeof(Button))]
+    public class ButtonSounds : MonoBehaviour
+    {
+        const float MinTimeBetweenSameClip = 0.1f;
+
+        // Direct interaction clips
+        public AudioClip ButtonCancelled;
+        public AudioClip ButtonHeld;
+        public AudioClip ButtonPressed;
+        public AudioClip ButtonReleased;
+
+        // State change clips
+        public AudioClip ButtonObservation;
+        public AudioClip ButtonObservationTargeted;
+        public AudioClip ButtonTargeted;
+
+        private AudioSource audioSource;
+        private static string lastClipName;
+        private static float lastClipTime;
+
+        void Start ()
+        {
+            Button button = GetComponent<Button>();
+            button.OnButtonCancelled += OnButtonCancelled;
+            button.OnButtonHeld += OnButtonHeld;
+            button.OnButtonPressed += OnButtonPressed;
+            button.OnButtonReleased += OnButtonReleased;
+            button.StateChange += StateChange;
+
+            audioSource = GetComponent<AudioSource>();
+        }
+
+        void StateChange(Button.ButtonStateEnum newState)
+        {
+            switch (newState)
```

Comment:
What about our other cases?

Reply:
Currently the ButtonSounds script and button profile only handle the primary interaction states and do not account for non-interaction states.